src/connecpy/async_client.py

```diff
@@ -97,26 +98,28 @@ async def _make_request(
                     url=self._address + url, content=request_data, **kwargs
                 )
 
-            resp.raise_for_status()
-
             if resp.status_code == 200:
                 response = response_class()
-                response.ParseFromString(resp.content)
+                try:
+                    response.ParseFromString(resp.content)
+                except Exception as e:
+                    raise exceptions.ConnecpyException(
+                        f"Failed to parse response message: {str(e)}"
+                    )
                 return response
             else:
-                raise exceptions.ConnecpyServerException.from_json(await resp.json())
+                raise ConnectWireError.from_dict(
+                    resp.json(), resp.status_code
+                ).to_exception()
         except httpx.TimeoutException as e:
             raise exceptions.ConnecpyServerException(
                 code=errors.Errors.DeadlineExceeded,
                 message=str(e) or "request timeout",
             )
-        except httpx.HTTPStatusError as e:
-            raise exceptions.ConnecpyServerException(
-                code=errors.Errors.Unavailable,
-                message=str(e),
-            )
+        except exceptions.ConnecpyException:
+            raise
         except Exception as e:
             raise exceptions.ConnecpyServerException(
-                code=errors.Errors.Internal,
+                code=errors.Errors.Unavailable,
```

Comment:
Unhandled exceptions here will be something like connection refused, etc. They should always be `Unavailable`, similar to connect-go here

https://github.com/connectrpc/connect-go/blob/84c176191838c0d5b4704840b6aa4a7f38dce0b9/duplex_http_call.go#L315